modules/jetty-http-server/src/test/java/com/spotify/apollo/http/server/ApolloRequestHandlerTest.java

```diff
@@ -141,6 +168,40 @@ public void shouldHandleMissingCallingServiceHeader() throws Exception {
                is(Optional.empty()));
   }
 
+  // note: this test may fail when running in IntelliJ, due to
+  // https://youtrack.jetbrains.com/issue/IDEA-122783
+  @Test
+  public void shouldLogErrorWritingResponse() throws Exception {
+    when(asyncContext.getResponse()).thenReturn(response);
+    when(response.getOutputStream()).thenReturn(outputStream);
+    doThrow(new IOException("expected")).when(outputStream).write(any(byte[].class));
+
+    ApolloRequestHandler.AsyncContextOngoingRequest ongoingRequest = new ApolloRequestHandler.AsyncContextOngoingRequest(
+        new ServerInfo() {
+          @Override
+          public String id() {
+            return "14";
+          }
+
+          @Override
+          public InetSocketAddress socketAddress() {
+            return InetSocketAddress.createUnresolved("localhost", 888);
+          }
+        },
+        Request.forUri("http://localhost:888"),
+        asyncContext, 9123
+    );
+
+    ongoingRequest.reply(Response.forPayload(ByteString.encodeUtf8("floop")));
+
+    List<LoggingEvent> events = testLogger.getAllLoggingEvents().stream()
+        .filter(event -> event.getLevel() == Level.ERROR)
+        .filter(event -> event.getMessage().contains("Failed to write response"))
+        .collect(Collectors.toList());
+
+    assertThat(events.size(), is(1));
```

Comment:
Could use org.hamcrest.Matchers.hasSize (or even org.hamcrest.Matchers.contains) for sightly better error message